plugins/syntax/g_numeral.ml4

```diff
@@ -0,0 +1,491 @@
+(************************************************************************)
+(*  v      *   The Coq Proof Assistant  /  The Coq Development Team     *)
+(* <O___,, *   INRIA - CNRS - LIX - LRI - PPS - Copyright 1999-2016     *)
+(*   \VV/  **************************************************************)
+(*    //   *      This file is distributed under the terms of the       *)
+(*         *       GNU Lesser General Public License Version 2.1        *)
+(************************************************************************)
+
+DECLARE PLUGIN "numeral_notation_plugin"
+
+open Pp
+open Util
+open Names
+open Libnames
+open Globnames
+open Constrexpr
+open Constr
+open Misctypes
+
+(** * Numeral notation *)
+
+(** Reduction
+
+    The constr [c] below isn't necessarily well-typed, since we
+    built it via an [mkApp] of a conversion function on a term
+    that starts with the right constructor, but might be partially
+    applied.
+
+    At least [c] is known to be evar-free, since it comes for
+    our own ad-hoc [constr_of_glob] or from conversions such
+    as [coqint_of_rawnum].
+*)
+
+let eval_constr (c : Constr.t) =
+  let env = Global.env () in
+  let j = Typeops.infer env c in
+  let c'=
+    Vnorm.cbv_vm env (Evd.from_env env)
+                 (EConstr.of_constr j.Environ.uj_val)
+                 (EConstr.of_constr j.Environ.uj_type)
+  in EConstr.Unsafe.to_constr c'
+
+(* For testing with "compute" instead of "vm_compute" :
+let eval_constr (c : constr) =
+ let (sigma, env) = Lemmas.get_current_context () in
+ Tacred.compute env sigma c
+*)
+
+let eval_constr_app c1 c2 = eval_constr (mkApp (c1,[| c2 |]))
+
+exception NotANumber
+
+let warning_big_num ty =
+  strbrk "Stack overflow or segmentation fault happens when " ++
+  strbrk "working with large numbers in " ++ pr_reference ty ++
+  strbrk " (threshold may vary depending" ++
+  strbrk " on your system limits and on the command executed)."
+
+let warning_abstract_num ty f =
+  let (sigma, env) = Pfedit.get_current_context () in
+  strbrk "To avoid stack overflow, large numbers in " ++
+  pr_reference ty ++ strbrk " are interpreted as applications of " ++
+  Printer.pr_constr_env env sigma f ++ strbrk "."
+
+(** Comparing two raw numbers (base 10, big-endian, non-negative).
+    A bit nasty, but not critical : only used to decide when a
+    nat number is considered as large. *)
```

Comment:
Why have a function only used for nat numbers? Or is this doc out of date and it should not say "nat"?

Reply:
Indeed, the warning infrastructure on "large" numbers could be used on any numeral type, not just `nat`. Even if `nat` is probably the only type really concerned by this feature. I've updated this comment.